skippy/state.go

```diff
@@ -87,6 +88,19 @@ func (s *State) SetThread(threadID string, thread *chatThread) {
 	s.threadMap[threadID] = thread
 }
 
+// used for locking a specific chat Thread
+// does not lock state
+// CALLER MUST CALL UnLockThread
+func (s *State) LockThread(threadID string) {
```

Comment:
this allows locking a particular chat thread so that the bot can handle quick message sends